### PR TITLE
REL-4279: Add basic Igrins2 support

### PR DESCRIPTION
**Pull request by cquiroz**
This will likely need refinements and we don't yet have an ICD but we can load and show Igrins2 sequences

image (13)

### CODE REVIEW

File: app/seqexec-server-gs/src/main/resources/app.conf [hunk 1]
```diff
@@ -62,7 +62,7 @@ seqexec-engine {
   ioTimeout = 4 seconds
   dhsTimeout = 30 seconds
   gpiUrl = "failover:(tcp://172.17.107.50:61616)?timeout=4000"
-  gpiGDS = "http://172.17.107.50:8888/xmlrpc"
```
Comment (framos-gemini) [inline app/seqexec-server-gs/src/main/resources/app.conf:65]: Just out of curiosity, Does this imply a change in the gpisimulator.cpp and gpiobssimulator.cpp examples of the giapi-glue-c++?

File: modules/server/src/main/scala/seqexec/server/igrins2/Igrins2Header.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  def header[F[_]: MonadThrow: Logger](
+    gdsClient:         GdsClient[F],
+    tcsKeywordsReader: TcsKeywordsReader[F]
+  ): Header[F] =
+    new Header[F] {
+      override def sendBefore(obsId: Observation.Id, id: ImageFileId): F[Unit] = {
+        val ks = GdsInstrument.bundleKeywords[F] {
+          List(
+            buildInt32(tcsKeywordsReader.igrins2InstPort, KeywordName.INPORT),
+            buildDouble(tcsKeywordsReader.p2ara, KeywordName.P2ARA),
```
Comment (jluhrs) [inline modules/server/src/main/scala/seqexec/server/igrins2/Igrins2Header.scala:26]: PWFS2 keywords are already set in `StandardHeader`, although only if `guideWithPWFS2` is active.

File: modules/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala [hunk 3]
```diff
@@ -455,6 +459,29 @@ trait TcsEpics[F[_]] {
     case G4 => g4GuideConfig
   }
 
+  def p2ara: F[Double]
```
Comment (jluhrs) [inline modules/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala:462]: These are already defined in `val pwfs2Target: Target[F]`

File: modules/server/src/main/scala/seqexec/server/tcs/TcsKeywordReader.scala [hunk 4]
```diff
@@ -171,8 +171,36 @@ trait TcsKeywordsReader[F[_]] {
 
   def ghostInstPort: F[Int]
 
+  def igrins2InstPort: F[Int]
+
   def crFollow: F[Option[CRFollow]]
 
+  def p2ara: F[Double]
```
Comment (jluhrs) [inline modules/server/src/main/scala/seqexec/server/tcs/TcsKeywordReader.scala:178]: They are already defined inside `def pwfs2Target: TargetKeywordsReader[F]`

File: modules/web/client/src/main/scala/seqexec/web/client/components/SeqexecMain.scala [hunk 5]
```diff
@@ -36,7 +36,8 @@ object AppTitle {
               horizontal = true,
               clazz = SeqexecStyles.titleRow |+| SeqexecStyles.notInMobile |+| SeqexecStyles.header
       )(
-        s"Seqexec ${p.site.shortName}",
+        s"Seqexec ${p.site.shortName}", // TODO: Remove me
+        <.div(^.cls := "igrins2-only", "IGRINS2 ONLY!"),
```
Comment (jluhrs) [inline modules/web/client/src/main/scala/seqexec/web/client/components/SeqexecMain.scala:40]: This should not be merged in main

File: modules/web/server/src/main/resources/logback.xml [hunk 6]
```diff
@@ -16,9 +16,9 @@
     <then>
         <!-- Files are rotated daily for up to 90 days -->
         <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
-            <file>/gemsoft/var/log/seqexec-ng/seqexec-ng.log</file>
+            <file>/home/software/logs/seqexec-ng.log</file>
             <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
-                <fileNamePattern>/gemsoft/var/log/seqexec-ng/seqexec-ng.%d{yyyy-MM-dd}.log</fileNamePattern>
+                <fileNamePattern>/home/softawre/logs/seqexec-ng.%d{yyyy-MM-dd}.log</fileNamePattern>
```
Comment (jluhrs) [inline modules/web/server/src/main/resources/logback.xml:21]: This and line 19 should not be merged into main

**Review by jluhrs — CHANGES_REQUESTED** [review]
It looks OK, but the whole issue with P2 keywords must be resolved.

File: modules/server/src/main/scala/seqexec/server/OdbProxy.scala [hunk 7]
```diff
@@ -159,15 +159,18 @@ object OdbProxy {
...
-        // We'll survive exceptions at the level of connecting to the wdba
-        L.warn(e.getMessage) *> List.empty.pure[F]
-      }
+      L.debug("Polling ODB for queued sequences") *>
+        F.delay(
+          xmlrpcClient
+            .getObservations(sessionName)
+            .toList
+            .flatMap(id => Observation.Id.fromString(id).toList)
+        ).flatTap(ids => L.debug(s"ODB poll returned: $ids"))
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/server/src/main/scala/seqexec/server/OdbProxy.scala:168]: I'll convert them to trace, they won't show on the log but can be activated if needed

File: modules/web/server/src/main/resources/logback.xml (hunk 6 above)
Comment (cquiroz) [reply modules/web/server/src/main/resources/logback.xml:21]: Solved

File: modules/server/src/main/resources/Tcs.xml [hunk 8]
```diff
@@ -1800,56 +1810,6 @@
             <type>STRING</type>
             <description>P2 Object Name</description>
         </attribute>
-        <attribute name="p2ara">
```
Comment (jluhrs) [inline modules/server/src/main/resources/Tcs.xml:1803]: All these definitions are used to set the P2 target keywords! They must be restored.

**Review by jluhrs — APPROVED** [review]
LGTM

**Pull request merged**